### PR TITLE
Port macos support from flutter_blue with clang-format

### PR DESCRIPTION
**Pull request by tuna-f1sh**
Re: https://github.com/boskokg/flutter_blue_plus/pull/140

### CODE REVIEW

**Comment by chipweinberger** [comment]
@tuna-f1sh , im not sure what changes you made, but can you not copy the existing code from the iOS side? 

There are a few changes. yours vs ios

**Comment by chipweinberger** [comment]
I used this clang style

{ColumnLimit: 120, IndentWidth: 4, BreakBeforeBraces: Allman}

File: ios/Classes/FlutterBluePlusPlugin.h
Comment (tuna-f1sh) [inline ios/Classes/FlutterBluePlusPlugin.h:1]: I've merged the upstream but I don't really understand the difference? The updated iOS code in upstream was newer and so would have taken precedence merging. The only changed in the ios files is this switch case for macOS.

**Comment by chipweinberger** [comment]
sorry if i'm clear. 

step 1: copy the ios code. this should be its own commit with a single file in the diff.

step 2: apply macos changes in 2nd commit.

**Comment by chipweinberger** [comment]
thanks for asking!

> So are you asking to review any new code in the iOS code (since pauldemarco/flutter_blue#626)

The diff from his branch is not important to me.

The main PR should be the diff vs current ios code.

Give me 1 sec, i'll do the first commit.

**Comment by chipweinberger** [comment]
Okay, you just need to rebase your changes now! Thats it! 

& please test to make sure macOS works. Thanks.

**Comment by tuna-f1sh** [comment]
Thanks. Unfortunately not so simple as a few of the new method fail to compile due to missing identifiers. I had a quite look:

* UIDevice name which I guess can be hostname or just returned as error as is iOS specific? I changed this to error for the ref below.
* didReadRSSI: needs more digging.

As you can probably tell, I'm not familiar enough with this code base to dig into it at the moment. I wanted to update this ticket to inform that.

ref
```
2023-05-01 16:32:45.266 xcodebuild[49736:10492109] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
/Users/john/git/flutter_blue_plus/macos/Classes/FlutterBluePlusPlugin.m:834:5: error: use of undeclared identifier 'ProtosReadRssiResult'
    ProtosReadRssiResult *result = [[ProtosReadRssiResult alloc] init];
    ^
/Users/john/git/flutter_blue_plus/macos/Classes/FlutterBluePlusPlugin.m:834:27: error: use of undeclared identifier 'result'
    ProtosReadRssiResult *result = [[ProtosReadRssiResult alloc] init];
                          ^
/Users/john/git/flutter_blue_plus/macos/Classes/FlutterBluePlusPlugin.m:834:38: error: use of undeclared identifier 'ProtosReadRssiResult'
    ProtosReadRssiResult *result = [[ProtosReadRssiResult alloc] init];
                                     ^
/Users/john/git/flutter_blue_plus/macos/Classes/FlutterBluePlusPlugin.m:835:6: error: use of undeclared identifier 'result'
    [result setRemoteId:[peripheral.identifier UUIDString]];
     ^
/Users/john/git/flutter_blue_plus/macos/Classes/FlutterBluePlusPlugin.m:836:6: error: use of undeclared identifier 'result'
    [result setRssi:[rssi intValue]];
     ^
/Users/john/git/flutter_blue_plus/macos/Classes/FlutterBluePlusPlugin.m:838:76: error: use of undeclared identifier 'result'
    [_channel invokeMethod:@"ReadRssiResult" arguments:[self toFlutterData:result]];
```

**Comment by chipweinberger** [comment]
oops I accidentally clicked the merge button.

> I'm not familiar enough with the code base

I was thinking we would just copy the changes from his branch _exactly_.

If there is an in implemented method, yes we can just leave it unimplemented or commented out for now.

**Comment by chipweinberger** [comment]
okay, I undid the merge

**Comment by tuna-f1sh** [comment]
> I was thinking we would just copy the changes from his branch exactly.

Now I'm really confused as what that preparation commit was for in that case! Anyway, I've made another PR https://github.com/boskokg/flutter_blue_plus/pull/269 with a single commit based on current upstream.

**Comment by chipweinberger** [comment]
Communication is hard....

The preparation commit is so that we can see & commit the diff between iOS and Mac! 

Basically, shouldn't it be possible to compare his code and our code, and copy all his changes manually?

**Pull request merged**